### PR TITLE
Salesforce to JSON - Contacts

### LINKED ISSUE #118: Customers export from Salesforce (JSON)
1. Read customer data from Salesforce
2. Save as local JSON file

---
Parameters
- JSON file location
- Salesforce creds

### PR DESCRIPTION
**Pull request by jannejjj**
Re-opened this since I had messed up the files in the previous one (created branch from another development branch instead of main which meant that this branch had 2 templates instead of one)

Implemented this so that it's the contacts that are exported instead of customers, since the import was done for contacts as well. The reasoning behind this is that the customer object type is quite restrictive. Another template can be made for exporting customers as well, or I can change this one. Open to comments!

closes #118 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
  - Introduced a new template process to export Salesforce contacts as a local JSON file, including detailed configurations, prerequisites, and error handling.

### CODE REVIEW

File: Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Implementation and Usage Notes
+
+This template fetches contacts from Salesforce and stores the information into a local JSON file.
+Process variables include fields for the Salesforce credentials and the output file path. If a file with the same name already exists, by default the process will throw an error - this behavior can be changed in the task settings.
+
+# Error Handling
+
+This template checks for errors after attempting to get the contact information from Salesforce task.
+If transient errors are expected, retries for the file write and Salesforce connections can be configured in the corresponding tasks.
```
[the remark this answers is not in the document]
Comment (jannejjj) [reply Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md:20]: Will fix.

**Comment by charlotta-frends** [comment]
The template seems to work as it should and looks good to me! Just a couple of comments/questions:

1. Contacts gotten successfully? - just a suggestion, but could something like "retrieved" sound more professional instead of "gotten"?
2. Should the template have dashed boxes? The process is quite small and easy to understand as it is, so I'm not sure if they are really necessary here, but thought I'd mention it just in case they need to be added.

**Review by charlotta-frends — APPROVED** [review]

File: Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+This template process retrieves the list of contacts from Salesforce and writes the information as a JSON file to the local fileshare.
+Using this template, you can retrieve the list of contacts from Salesforce and store it locally as JSON.
```
Comment (jefim) [inline Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md:2]: double newline

File: Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Prerequisites
+
+This template assumes that the following prerequisites are in place:
+
+- The Salesforce user should have access to the client ID, secret and security token.
+
+# Implementation and Usage Notes
+
+This template fetches contacts from Salesforce and stores the information into a local JSON file.
+Process variables include fields for the Salesforce credentials and the output file path. If a file with the same name already exists, by default the process will throw an error - this behavior can be changed in the task settings.
```
Comment (jefim) [inline Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md:15]: double newline

File: Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+# Error Handling
+
+This template checks for errors after attempting to get the contact information from Salesforce.
+If transient errors are expected, retries for the file write and Salesforce connections can be configured in the corresponding tasks.
```
Comment (jefim) [inline Templates/CRMs/Salesforce/Salesforce to JSON - Contacts/long-description.md:20]: double newline

**Review by jefim — CHANGES_REQUESTED** [review]

**Comment by charlotta-frends** [comment]
This is ready to be merged!

**Review by jefim — APPROVED** [review]

**Pull request merged**